internal/application/workload.go

```diff
@@ -185,24 +156,53 @@ func (a *Workload) Replicas(ctx context.Context) (map[string]*models.PodInfo, er
 // Get returns the state of the app deployment encoded in the workload.
 func (a *Workload) Get(ctx context.Context) (*models.AppDeployment, error) {
 
-	deployment, err := a.Deployment(ctx)
+	// Information about the active workload.
+	// The data is pulled out of the list of Pods associated with the application.
+	// The originally asked the app's Deployment resource. With app charts the existence
+	// of such a resource cannot be guarantueed any longer.
+
+	podList, err := a.Pods(ctx)
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return nil, err
-		}
-		// App is inactive, no deployment, no workload
+		return nil, err
+	}
+
+	// No pods => no workload
+	if len(podList) == 0 {
 		return nil, nil
 	}
 
-	desiredReplicas := deployment.Status.Replicas
-	readyReplicas := deployment.Status.ReadyReplicas
+	var readyReplicas int32
+	var createdAt time.Time
+	var stageID string
+	var username string
+	var controllerName string
+
+	if len(podList) > 0 {
+		// Pods found. replace defaults with actual information.
+
+		// Initialize various pieces from the first pod ...
+		createdAt = podList[0].ObjectMeta.CreationTimestamp.Time
```

Comment:
This won't reflect reality anymore. A Deployment may be scaled up or down, causing Pods to appear and disappear. The age of any Pod may have nothing to do with the age of the "workload". That said, the age of the workload is probably nonsense. How old is the "workload" of an application that run 1 Pod for 3 hours, then was scaled down to 0 for an hour or two and then scaled back up to 3 Pods for 10 hours? If we imply that the age of the workload is the time since any Pod was created, then in most cases what we need is the time of creation of the application itself, unless the application was originally created without a workload and stayed like that for while. In that case though, I don't think the creation date of the first Pod is useful in any way.

Unless I'm missing something and this breaks the cli or the UI in a very bad way, I suggest we remove this field from the workload. This information should be provided for the individual Pods.

Reply:
I believe you missed the code later in the loop which merges the dates from all the pods, and takes the youngest.
Lines 190-193.

Regarding usage ... The cli uses it for the general `Age` information, see (***) below

```
|         KEY          |                    VALUE                     |
|----------------------|----------------------------------------------|
| Origin               | /home/work/SUSE/dev/epinio/assets/sample-app |
| Created              | 2022-07-26 15:01:14 +0200 CEST               |
| Status               | 1/1                                          |
| Username             | admin                                        |
| Running StageId      | e6b05fbad294fc4b                             |
| Last StageId         | e6b05fbad294fc4b                             |
| Age                  | 6m0s                                         | ***
| Active Routes        |                                              |
|                      | fox-a279f.172.19.0.4.omg.howdoi.website      |
| App Chart            | standard                                     |
| Desired Instances    |                                            1 |
| Bound Configurations |                                              |
| Environment          |                                              |
```

The cli also show per-pod age.
No idea about usage in the UI . @richard-cox ping

Note that a full removal of the field from the structure is an API change now.

For best compatibility we should keep the field, even if we decide to not fill it anymore, i.e. leave it at Go's zero value.

Reply:
Yes I missed that part but my comment still stands, no Pod age is guaranteed to match a "Deployment age" whatever the latter means.

Reply:
I might have got confused with the terminology here, but the UI presents app, deployment and 'instances' info as below

![image](https://user-images.githubusercontent.com/18697775/181018085-5790e6f9-2706-457d-9002-e27bcbbf044c.png)

- App age at the top (from `meta.createdAt`)
- No deployment age (though there would be value in showing, from `deployment.createdAt`)
- 'Instances' Age per row. (from `deployment.replicas` and their `createdAt` property)

I would be nice to keep all of the above, unless the replicaSet age won't differ between entries

Reply:
@richard-cox 
The `deployment age` was from the `Deployment` resource of the app. With us not knowing anymore if the top controller is a `Deployment`/`StatefulSet`/other I am currently trying to synthesize something sensible from the App Pods (I take the ... oldest Pod as the reference) ... @jimmykarily only saw me taking the first Pod.

Reply:
> match a "Deployment age" whatever the latter means.

The "Deployment age" was literally the `Deployment`'s `MetaData.CreatedAt`.

Does the go kube client stuff have a general way of getting a resource's meta data without knowing the kind of the resource ? I.e. just from the name+namespace ?

If yes, then we can get rid of the code synthesizing something from the pods in favor of pulling from the controller resource again. Note, at that point the new `epinio.io/app-container` label is also important as naming the controller resource.
We could of course separate this and have a `epinio.io/controller` label as well.

Reply:
... Oh, it might be do-able via `unstructured`

Reply:
There is a way to discover what controller the Pod belongs too but I'm not sure if it's guaranteed (e.g. what happens if the application chart simply creates Pods without Deployments or Statefulsets?).

The Pod has an owner field. In the case of Deployments that is a ReplicaSet. The ReplicaSet has an owner, which is the Deployment. I assume there is a similar chain of owners for a Statefulset. We may not even have to go that up, maybe the createdAt field of the first Owner in the chain is enough.

If there is no owner (e.g. a custom application chart creates just Pods for whatever reason), we could opt for the current solution.

I wonder if it's too complicated logic, to get information that makes little sense imho.

Reply:
Or not ... Seems even the dynamic client wants me to tell it the kind of resource ahead of time ... I guess that is because a resource is identified uniquely only by all three of (name, namespace, and kind).

In that case case the chart would have to tell Epinio the kind, another label on the pods.

Reply:
Ok. Stopping here. Putting this all on #1657 ...